### PR TITLE
Do not compare reference values across executions

### PR DESCRIPTION
**Pull request by tlively**
Since we optimize assuming a closed world, optimizations can change the types
and structure of GC data even in externally-visible ways. Because differences
are expected, the fuzzer already did not compare reference-typed values from
before and after optimizations when running with nominal typing. Update it to
not compare these values under any type system.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5276** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5276" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5276?utm_source=stack-comment).

**Review by kripken — APPROVED** [review]

**Comment by tlively** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5276?utm_source=stack-comment).


Job Reference: lJAMT4rkCAqvOwNbZeQD

**Pull request merged**
